File: abc/abc018/abc018c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func f(cy, cx, K int, t [][]int) bool {
	for y := cy - (K - 1); y < cy+K; y++ {
		l := cx - (K - 1) + abs(y-cy)
		r := cx + (K - 1) - abs(y-cy)
		if l == 0 {
			if t[y][r] != 0 {
				return false
			}
		} else {
			if t[y][r]-t[y][l-1] != 0 {
				return false
			}
		}
	}
	return true
}

func main() {
	defer flush()

	R := readInt()
	C := readInt()
	K := readInt()

	s := make([]string, R)
	for i := 0; i < R; i++ {
		s[i] = readString()
	}

	t := make([][]int, R)
	for i := 0; i < R; i++ {
		t[i] = make([]int, C)
	}

	for y := 0; y < R; y++ {
		for x := 0; x < C; x++ {
			if s[y][x] == 'x' {
				t[y][x] = 1
			}
			if x != 0 {
				t[y][x] += t[y][x-1]
			}
		}
	}

	result := 0
	for y := K - 1; y < R-(K-1); y++ {
		for x := K - 1; x < C-(K-1); x++ {
			if f(y, x, K, t) {
				result++
			}
		}
	}
	println(result)
}

const (
	ioBufferSize = 1 * 1024 * 1024 // 1 MB
)

var stdinScanner = func() *bufio.Scanner {
	result := bufio.NewScanner(os.Stdin)
	result.Buffer(make([]byte, ioBufferSize), ioBufferSize)
	result.Split(bufio.ScanWords)
	return result
}()

func readString() string {
	stdinScanner.Scan()
	return stdinScanner.Text()
}

func readInt() int {
	result, err := strconv.Atoi(readString())
	if err != nil {
		panic(err)
	}
	return result
}

var stdoutWriter = bufio.NewWriter(os.Stdout)

func flush() {
	stdoutWriter.Flush()
}

func println(args ...interface{}) (int, error) {
	return fmt.Fprintln(stdoutWriter, args...)
}

File: abc/abc018/abc018c.py
from itertools import accumulate


def f(cy, cx, K, t):
    for y in range(cy - (K - 1), cy + K):
        l = cx - (K - 1) + abs(y - cy)
        r = cx + (K - 1) - abs(y - cy)
        if l == 0:
            if t[y][r] != 0:
                return False
        else:
            if t[y][r]-t[y][l-1] != 0:
                return False
    return True


R, C, K = map(int, input().split())
s = [input() for _ in range(R)]

t = [[0] * C for _ in range(R)]
for y in range(R):
    for x in range(C):
        if s[y][x] == 'x':
            t[y][x] = 1
    t[y] = list(accumulate(t[y]))

result = 0
for y in range(K - 1, R - (K - 1)):
    for x in range(K - 1, C - (K - 1)):
        if f(y, x, K, t):
            result += 1
print(result)
